org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/SolsticeScanMonitor.java

```diff
@@ -0,0 +1,236 @@
+		} catch (Exception ne) {
+			logger.debug("Unable to write point models", ne);
+		}
+
+		// create the scan finished dataset and set the initial value to false
+//		scanFinished = scanPointsCollection.initializeFixedSizeLazyDataset(
+//				FIELD_NAME_SCAN_FINISHED, new int[] { 1 }, Dataset.INT32);
+		// TODO: workaround for bug in HD5 loader, do not set size limit 
+		scanFinished = new LazyWriteableDataset(FIELD_NAME_SCAN_FINISHED, Integer.class,
+				new int[] { 1 }, new int[] { -1 }, new int[] { 1 }, null);
+		scanFinished.setFillValue(0);
+		scanPointsCollection.createDataNode(FIELD_NAME_SCAN_FINISHED, scanFinished);
+		
+		// create a sub-collection for the unique keys field and keys from each external file
+		final NXcollection keysCollection = NexusNodeFactory.createNXcollection();
+		scanPointsCollection.addGroupNode(GROUP_NAME_KEYS, keysCollection);
+		
+		// create the unique keys dataset (not for malcolm scans)
+		if (!malcolmScan) {
+			uniqueKeys = keysCollection.initializeLazyDataset(FIELD_NAME_UNIQUE_KEYS, info.getRank(), Integer.class);
+		}
+
+		// set chunking for lazy datasets
+		if (info.getRank() > 0) {
+			final int[] chunk = info.createChunk(false, 8);
+			if (!malcolmScan) {
+				uniqueKeys.setFillValue(0);
+				uniqueKeys.setChunking(chunk);
+			}
+		}
+		
+		// add external links to the unique key datasets for each external HD5 file
+		addLinksToExternalFiles(keysCollection);
+		
+		return scanPointsCollection;
+	}
+
+	/**
+	 * Called when the scan completes to write '1' to the scan finished dataset.
+	 * @throws ScanningException
+	 */
+	public void scanFinished() throws ScanningException {
+		// Note: we don't use IRunListener.runPerformed as other run listeners expect the
+		// file to be closed when that method is called
+		final Dataset dataset = DatasetFactory.createFromObject(IntegerDataset.class, 1, null);
+		try {
+			this.scanFinished.setSlice(null, dataset,
+					new int[] { 0 }, new int[] { 1 }, new int[] { 1 });
+		} catch (Exception e) {
+			throw new ScanningException("Could not write scanFinished to NeXus file", e);
+		}
+	}
+
+	@Override
+	public Object getPosition() throws Exception {
+		return null;
+	}
+
+	@Override
+	public void setPosition(Object value, IPosition position) throws Exception {
+		writePosition(position);
+	}
+
+	/**
+	 * For each device, if that device writes to an external file, create an external link
+	 * within the unique keys collection to the unique keys dataset in that file.
+	 * The unique keys are required in order for live processing to take place - we need to know
+	 * how much data has been written to each file - devices may flush their data to file at
+	 * different times.
+	 * @param uniqueKeysCollection unique keys collection to add any external links to
+	 */
+	private void addLinksToExternalFiles(final NXcollection uniqueKeysCollection) {
+		if (nexusObjectProviders == null) throw new IllegalStateException("nexusObjectProviders not set");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)